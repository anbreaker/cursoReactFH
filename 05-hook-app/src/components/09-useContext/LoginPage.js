import React, {useContext} from 'react';
import {UserContext} from './UserContext';

export const LoginPage = () => {
  const {setUser} = useContext(UserContext);

  return (
    <div>
      <div>
        <h1>LoginPage</h1>
        <hr />

        <button
          className="btn btn-primary"
          onClick={() => setUser({id: 123, name: 'anbreaker'})}>
          Login
        </button>
      </div>
    </div>
  );
};
