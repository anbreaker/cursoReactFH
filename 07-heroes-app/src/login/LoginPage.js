import React from 'react';

export const LoginPage = ({history}) => {
  const handleClick = () => {
    // history.push('/');
    history.replace('/');
  };
  return (
    <div className="container">
      <h1>LoginPage</h1>
      <hr />

      <button className="btn btn-primary" onClick={handleClick}>
        Login
      </button>
    </div>
  );
};
