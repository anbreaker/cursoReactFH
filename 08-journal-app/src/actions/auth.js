import {firebase, googleAuthProvider} from '../firebase/firebaseConfig';
import {types} from '../types/types';

export const startLoginEmailPassword = (email, password) => {
  return (dispatch) => {
    firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(({user}) => {
        dispatch(login(user.uid, user.displayName));
      })
      .catch((error) => {
        console.error('Error ->', error);
      });
  };
};

export const startRegisterWithEmailPasswordName = (email, password, name) => {
  return (dispatch) => {
    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(async ({user}) => {
        await user.updateProfile({displayName: name});

        dispatch(login(user.uid, user.displayName));
      })
      .catch((error) => {
        console.error('Error ->', error);
      });
  };
};

export const startGoogleLogin = () => {
  return (dispatch) => {
    firebase
      .auth()
      .signInWithPopup(googleAuthProvider)
      .then(({user}) => {
        dispatch(login(user.uid, user.displayName));
      });
  };
};

export const login = (uid, displayName) => {
  return {
    type: types.login,
    payload: {
      uid,
      displayName,
    },
  };
};
