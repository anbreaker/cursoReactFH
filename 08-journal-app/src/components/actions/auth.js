import {firebase, googleAuthProvider} from '../../firebase/firebaseConfig';
import {types} from '../../types/types';

export const startEmailPassword = (email, password) => {
  return (dispatch) => {
    setTimeout(() => {
      dispatch(login(123, 'curro'));
    }, 3000);
  };
};

export const startGoogleLogin = () => {
  return (dispatch) => {
    firebase
      .auth()
      .signInWithPopup(googleAuthProvider)
      .then(({user}) => {
        dispatch(login(user.uid, user.displayName));
      });
  };
};

export const login = (uid, displayName) => {
  return {
    type: types.login,
    payload: {
      uid,
      displayName,
    },
  };
};
