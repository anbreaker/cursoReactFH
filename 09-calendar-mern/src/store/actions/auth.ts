import Swal from 'sweetalert2';
import 'sweetalert2/dist/sweetalert2.css';

import { fetchWithoutToken, fetchWithToken } from '../../helpers/fetch';
import { types } from '../../types/types';
import { eventLogout } from './events';

export const startLogin = (email: string, password: string) => {
  return async (dispatch: any) => {
    const response = await fetchWithoutToken('auth', { email, password }, 'POST');

    const body = await response.json();

    if (body.ok) {
      loginOrRegisterFetch(body);

      dispatch(
        login({
          uid: body.uid,
          name: body.name,
        })
      );
    } else {
      Swal.fire('Error', 'Usuario o contraseÃ±ar Erroneo.', 'error');
    }
  };
};

const login = (user: any) => ({
  type: types.authLogin,
  payload: user,
});

export const startRegister = (name: string, email: string, password: string) => {
  return async (dispatch: any) => {
    const response = await fetchWithoutToken(
      'auth/new',
      { name, email, password },
      'POST'
    );

    const body = await response.json();

    if (body.ok) {
      loginOrRegisterFetch(body);

      dispatch(
        login({
          uid: body.uid,
          name: body.name,
        })
      );
    } else {
      Swal.fire('Error', body.msg, 'error');
    }
  };
};

export const startChecking = () => {
  return async (dispatch: any) => {
    const response = await fetchWithToken('auth/renew');

    const body = await response.json();

    if (body.ok) {
      loginOrRegisterFetch(body);

      dispatch(
        login({
          uid: body.uid,
          name: body.name,
        })
      );
    } else {
      dispatch(checkingFinish());
    }
  };
};

export const startLogout = () => {
  return (dispatch: any) => {
    localStorage.clear();

    dispatch(logout());

    dispatch(eventLogout());
  };
};

const logout = () => ({ type: types.authLogout });

const checkingFinish = () => ({ type: types.authCheckingFinish });

const loginOrRegisterFetch = (body: any) => {
  localStorage.setItem('token', body.token);
  localStorage.setItem('token-init-date', new Date().getTime().toString());
};
