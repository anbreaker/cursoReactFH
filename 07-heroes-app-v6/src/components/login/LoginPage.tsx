import { useNavigate } from 'react-router-dom';
import { useContext } from 'react';

import { ActionsTypes } from '../../types/actions';
import { AuthContext } from '../../auth/authContext';

export const LoginPage = () => {
  const navigate = useNavigate();

  const { dispatch }: any = useContext(AuthContext);

  const handleLogin = () => {
    const action = {
      type: ActionsTypes.LOGIN,
      payload: {
        name: 'anbreaker',
      },
    };

    dispatch(action);

    const lastPath = localStorage.getItem('lastPath');

    navigate(lastPath || '/marvel', {
      replace: true,
    });
  };

  return (
    <div className="container mt-5">
      <h1>LoginPage</h1>
      <hr />

      <button className="btn btn-primary" onClick={handleLogin}>
        Login
      </button>
    </div>
  );
};
