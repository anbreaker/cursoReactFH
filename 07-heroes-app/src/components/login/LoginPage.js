import React, {useContext} from 'react';
import {AuthContext} from '../../auth/AuthContext';
import {types} from '../../types/types';

export const LoginPage = ({history}) => {
  const {dispatch} = useContext(AuthContext);

  const handleClick = () => {
    history.push('/');
    // history.replace('/');

    const action = {
      type: types.login,
      payload: {
        name: 'anbreaker',
      },
    };

    dispatch(action);
  };
  return (
    <div className="container">
      <h1>LoginPage</h1>
      <hr />

      <button className="btn btn-primary" onClick={handleClick}>
        Login
      </button>
    </div>
  );
};
