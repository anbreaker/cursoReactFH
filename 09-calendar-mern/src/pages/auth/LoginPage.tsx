import { useDispatch } from 'react-redux';
import { useForm } from '../../hooks/useForm';
import { startLogin, startRegister } from '../../store/actions/auth';
import Swal from 'sweetalert2';
import 'sweetalert2/dist/sweetalert2.css';

export const LoginPage = () => {
  const dispatch = useDispatch();

  const [formLoginValues, handleLoginInputChange]: any = useForm({
    loginEmail: 'test@test.com',
    loginPassword: 'test',
  });

  const { loginEmail, loginPassword }: any = formLoginValues;

  const [formRegisterValues, handleRegisterInputChange]: any = useForm({
    regName: 'anbreaker',
    regEmail: 'test@test.com',
    regPassword1: 'test',
    regPassword2: 'test',
  });

  const { regName, regEmail, regPassword1, regPassword2 }: any = formRegisterValues;

  const handleLogin = (event: any) => {
    event.preventDefault();

    dispatch(startLogin(loginEmail, loginPassword));
  };

  const handleRegister = (event: any) => {
    event.preventDefault();

    if (regPassword1 !== regPassword2)
      Swal.fire('Error', 'Las contrase単as deben coincidir', 'error');

    dispatch(startRegister(regName, regEmail, regPassword1));
  };

  return (
    <div className="container login-container">
      <div className="row">
        <div className="col-md-6 login-form-1">
          <h3>Ingreso</h3>
          <form onSubmit={handleLogin}>
            <div className="form-group">
              <input
                type="text"
                className="form-control"
                placeholder="Correo"
                name="loginEmail"
                value={loginEmail}
                onChange={handleLoginInputChange}
              />
            </div>
            <div className="form-group">
              <input
                type="password"
                className="form-control"
                placeholder="Contrase単a"
                name="loginPassword"
                value={loginPassword}
                onChange={handleLoginInputChange}
              />
            </div>
            <div className="form-group">
              <input type="submit" className="btnSubmit" value="Login" />
            </div>
          </form>
        </div>

        <div className="col-md-6 login-form-2">
          <h3>Registro</h3>
          <form onSubmit={handleRegister}>
            <div className="form-group">
              <input
                type="text"
                className="form-control"
                placeholder="Nombre"
                name="regName"
                value={regName}
                onChange={handleRegisterInputChange}
              />
            </div>
            <div className="form-group">
              <input
                type="email"
                className="form-control"
                placeholder="Correo"
                name="regEmail"
                value={regEmail}
                onChange={handleRegisterInputChange}
              />
            </div>
            <div className="form-group">
              <input
                type="password"
                className="form-control"
                placeholder="Contrase単a"
                name="regPassword1"
                value={regPassword1}
                onChange={handleRegisterInputChange}
              />
            </div>

            <div className="form-group">
              <input
                type="password"
                className="form-control"
                placeholder="Repita la contrase単a"
                name="regPassword2"
                value={regPassword2}
                onChange={handleRegisterInputChange}
              />
            </div>

            <div className="form-group">
              <input type="submit" className="btnSubmit" value="Crear cuenta" />
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};
