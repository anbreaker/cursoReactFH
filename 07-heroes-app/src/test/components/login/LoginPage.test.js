import {mount} from 'enzyme';
import {AuthContext} from '../../../auth/AuthContext';
import {LoginPage} from '../../../components/login/LoginPage';
import {types} from '../../../types/types';

describe('Pruebas en el componente <LoginPage />  ', () => {
  const history = {
    replace: jest.fn(),
  };

  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: false,
    },
  };

  const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
      <LoginPage history={history} />
    </AuthContext.Provider>
  );

  test('debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de realizar el dispatch y la navegacion', () => {
    const handleClick = wrapper.find('button').prop('onClick');
    handleClick();

    expect(contextValue.dispatch).toHaveBeenCalledWith({
      type: types.login,
      payload: {
        name: 'anbreaker',
      },
    });

    expect(history.replace).toHaveBeenCalledWith('/');

    localStorage.setItem('lastPath', '/dc');
    handleClick();
    expect(history.replace).toHaveBeenCalledWith('/dc');
  });
});
