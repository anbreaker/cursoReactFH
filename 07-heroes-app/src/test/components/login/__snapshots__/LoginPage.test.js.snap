// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <LoginPage />   debe mostrarse correctamente 1`] = `
<div
  className="container"
>
  <h1>
    LoginPage
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
